### ISSUE TITLE
Divide the hotspot_all test target into smaller targets 

### ISSUE THREAD

**Issue by smlambert** [AdoptOpenJDK/openjdk-tests#2477]
The [hotspot_all](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/openjdk/playlist.xml#L40-L66) target contains a lot of tests and could benefit from splitting up.

2 reasons it will be useful to divide hotspot_all test target:
- smaller targets will then allow for more granular rerunning/exclusion/dividing for running in PARALLEL
- This will also help resolve the request in https://github.com/eclipse/openj9/issues/12276#issuecomment-805253021 to add a [hotspot_serviceability](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/3c980f5a05aa5123fb6f965ea4c4139aea9cca01/test/hotspot/jtreg/TEST.groups#L55) target for openj9 for xlinux (as AsyncGetCallTrace support is being added to openj9).

We can use the [hotspot/jtreg/TEST.groups](https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/hotspot/test/TEST.groups) file as a guide to how to divide hotspot_all into subsets (without accidentally dropping some tests in the refactoring).  Based on the contents of TEST.groups, we could divide hotspot_all into [4 targets](https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/hotspot/test/TEST.groups#L145-L148), hotspot_compiler, hotspot_gc, hotspot_runtime and hotspot_serviceability targets.

**Comment by vanichitkara** [comment]
I'd like to work on this issue! Can it be assigned to me? Thanks!

**Comment by vanichitkara** [comment]
Hi @smlambert! Can you please elaborate about what all is to be done in the issue. Also do I have to create a new file for PR or edit the [hotspot_all](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/openjdk/playlist.xml#L40-L66) target file?

**Comment by smlambert** [comment]
What is needed for this issue is to edit the playlist.xml file.

Each test in that file gets converted into a make target.

Currently there is a test called hotspot_all in that file.  The hotspot_all target runs many tests.  We would like to divide the hotspot_test into 4 tests.  

You can see that the `<command>` tag in the hotspot_all test refers to some jtreg TEST.groups based on the TEST.groups, the natural way to divide hotspot_all is by making 4 new `<test>` blocks, one for each hotspot_compiler, hotspot_gc, hotspot_runtime and hotspot_serviceability.

So really the task is to remove 1 test block (hotspot_all) and add 4 blocks (hotspot_compiler, hotspot_gc_hotspot_runtime, hotspot_serviceability) to playlist.xml file (you can copy/paste the hotspot_all block and change 2 parts, the testCaseName and the command, like this:

```
	<test>
		<testCaseName>hotspot_compiler</testCaseName>
		<variations>
			<variation>Mode150</variation>
			<variation>Mode650</variation>
			<variation>Mode1000</variation>
		</variations>
		<command>$(JAVA_COMMAND) -Xmx512m -jar $(Q)$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar$(Q) \
	$(JTREG_BASIC_OPTIONS) $(JVM_NATIVE_OPTIONS) -vmoptions:$(Q)-Xmx512m $(JVM_OPTIONS)$(Q) \
	-w $(Q)$(REPORTDIR)$(D)work$(Q) \
	-r $(Q)$(TEST_RESROOT)$(D)report$(Q) \
	-jdk:$(Q)$(TEST_JDK_HOME)$(Q) \
	$(Q)$(JTREG_HOTSPOT_TEST_DIR):hotspot_compiler$(Q); \
	$(TEST_STATUS)</command>
		<versions>
			<version>8</version>
			<version>9</version>
		</versions>
		<levels>
			<level>extended</level>
		</levels>
		<groups>
			<group>openjdk</group>
		</groups>
		<impls>
			<impl>hotspot</impl>
		</impls>
	</test>
```

For the hotspot_serviceability target, we would remove the specific `<impls>` tag, because we want it to run against all impls, not just hotspot
```
	<test>
		<testCaseName>hotspot_serviceability</testCaseName>
		<variations>
			<variation>Mode150</variation>
			<variation>Mode650</variation>
			<variation>Mode1000</variation>
		</variations>
		<command>$(JAVA_COMMAND) -Xmx512m -jar $(Q)$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar$(Q) \
	$(JTREG_BASIC_OPTIONS) $(JVM_NATIVE_OPTIONS) -vmoptions:$(Q)-Xmx512m $(JVM_OPTIONS)$(Q) \
	-w $(Q)$(REPORTDIR)$(D)work$(Q) \
	-r $(Q)$(TEST_RESROOT)$(D)report$(Q) \
	-jdk:$(Q)$(TEST_JDK_HOME)$(Q) \
	$(Q)$(JTREG_HOTSPOT_TEST_DIR): hotspot_serviceability$(Q); \
	$(TEST_STATUS)</command>
		<versions>
			<version>8</version>
			<version>9</version>
		</versions>
		<levels>
			<level>extended</level>
		</levels>
		<groups>
			<group>openjdk</group>
		</groups>
	</test>
```

**Comment by vanichitkara** [comment]
Thank you so much @smlambert for explaining the task se well. Made a PR for resolving the issue, please review it.

**Comment by smlambert** [comment]
Closed via #2499
